### PR TITLE
Propogate previously unused NOTIFICATION_WORLD_2D_CHANGED, make CanvasItem/Camera2D/CollisionObject2D use it

### PR DESCRIPTION
**Pull request by spacechase0**
(Fixes godotengine/godot#52423)

I'm not sure how do this in the master branch. I see in the comment in the PR template saying to do things for master first - if I could get some guidance on that, I would appreciate it.

### CODE REVIEW

**Review by pouleyKetchoupp — CHANGES_REQUESTED** [review]
Thanks for contributing this fix!

It looks like it makes sense to handle world changes in 2D, although I haven't tested yet. I've left some comments on parts of the code that need to be a bit different from 3D logic, since entering/leaving worlds doesn't work the same in 2D and 3D.

Two other things that are needed before a possible approval:

- Documentation needs to be updated
Here for the new notification: https://github.com/godotengine/godot/blob/master/doc/classes/CanvasItem.xml#L508
Here for the new function call: https://github.com/godotengine/godot/blob/master/doc/classes/CanvasItem.xml#L19

- A master version of this PR is needed
What exactly was blocking you from making a master PR? Is about setting the branch, testing your changes, or a more specific issue with the code?

File: scene/2d/canvas_item.cpp
Comment (spacechase0) [reply scene/2d/canvas_item.cpp:605]: Just those lines you suggested did not work (on master, working on this now - opening a new PR soon). I'm not sure what in `_exit_canvas()`/`_enter_cavnas()` makes it work, but trying just the snippet in your comment made it where the circle under the character did not move with it graphically.

**Comment by spacechase0** [comment]
(Made a new PR that is for master, closing this one.)

**Pull request closed without merging**
